src/neuroconv/tools/nwb_helpers/_dataset_configuration.py

```diff
@@ -177,16 +177,15 @@ def get_default_dataset_io_configurations(
         if isinstance(neurodata_object, DynamicTable):
             dynamic_table = neurodata_object  # for readability
 
-            for column_name in dynamic_table.colnames:
-                candidate_dataset = dynamic_table[column_name].data  # VectorData object
```

Comment:
Wait, so how are these two guys differently?

Reply:
Yep 😅

Reply:
In particular, `dynamic_table[column_name]` calls `__get_item__` on `dynamic_table` with key `column_name`, which default `dict` behavior would return the value, but they adjust it to return downstream links so that you can do things like `units["spike_times"][:]` and it returns the actual list of spikes shaped by `units x spikes_per_unit`